source/d3d11/d3d11_device_context.cpp

```diff
@@ -16,23 +16,254 @@ void D3D11DeviceContext::log_drawcall(UINT vertices)
+	com_ptr<ID3D11Texture2D> depthstencil_texture;
+	depthStencilView->GetResource(&resource);
+	if (FAILED(resource->QueryInterface(&texture)))
+	{
+		return nullptr;
+	}
+	texture->GetDesc(&texdesc);
+
+	depthstencil_texture = texture.get();
+
+	_draw_call_tracker.set_depth_texture(depthstencil_texture.get());
+
+	D3D11_DEPTH_STENCIL_VIEW_DESC dsvdesc = {};
+	dsvdesc.ViewDimension = D3D11_DSV_DIMENSION_TEXTURE2D;
+
+	switch (texdesc.Format)
+	{
+		case DXGI_FORMAT_R16_TYPELESS:
+			dsvdesc.Format = DXGI_FORMAT_D16_UNORM;
+			break;
+		case DXGI_FORMAT_R32_TYPELESS:
+			dsvdesc.Format = DXGI_FORMAT_D32_FLOAT;
+			break;
+		case DXGI_FORMAT_R24G8_TYPELESS:
+			dsvdesc.Format = DXGI_FORMAT_D24_UNORM_S8_UINT;
+			break;
+		case DXGI_FORMAT_R32G8X24_TYPELESS:
+			dsvdesc.Format = DXGI_FORMAT_D32_FLOAT_S8X24_UINT;
+			break;
+		default:
+			dsvdesc.Format = texdesc.Format;
+			break;
+	}
+
+	HRESULT hr = _device->CreateDepthStencilView(texture.get(), &dsvdesc, &depthStencilView_copy);
+
+	if (FAILED(hr))
+	{
+		LOG(ERROR) << "Failed to create depth stencil view copy! HRESULT is '" << std::hex << hr << std::dec << "'.";
+		return nullptr;
+	}
+
+	return depthStencilView_copy;
+}
+
+bool D3D11DeviceContext::check_depthstencil(com_ptr<ID3D11Texture2D> texture, D3D11_TEXTURE2D_DESC texture_desc)
+{	
+	screen_dimensions();
+	float aspect_ratio = ((float)_screen_width) / ((float)_screen_height);
+	float twfactor = 1.0f;
+	float thfactor = 1.0f;
+
+	// check the depthstencil flag
+	if (texture_desc.BindFlags != (D3D11_BIND_DEPTH_STENCIL | D3D11_BIND_SHADER_RESOURCE) && texture_desc.BindFlags != D3D11_BIND_DEPTH_STENCIL)
+	{
+		return false;
+	}
+
+	if (texture_desc.Width != _screen_width)
+	{
+		twfactor = (float)_screen_width / texture_desc.Width;
+	}
+	if (texture_desc.Height != _screen_height)
+	{
+		thfactor = (float)_screen_height / texture_desc.Height;
+	}
+
+	// check if aspect ratio is similar to the back buffer one
+	float stencilaspectratio = ((float)texture_desc.Width) / ((float)texture_desc.Height);
+	if (fabs(stencilaspectratio - aspect_ratio) > 0.1f)
+	{
+		return false;
+	}
+
+	return true;
+}
+
 void D3D11DeviceContext::set_active_depthstencil(ID3D11DepthStencilView* pDepthStencilView)
 {
-	if (pDepthStencilView != _active_depthstencil)
+	if (pDepthStencilView != nullptr)
+	{
+		if (pDepthStencilView != _active_depthstencil)
+		{
+			_active_depthstencil = pDepthStencilView;
+			_draw_call_tracker.track_depthstencil(_active_depthstencil.get());
+		}
+	}
+}
+
+void D3D11DeviceContext::set_active_OM_depthstencil(ID3D11DepthStencilView* pDepthStencilView)
+{
+	if (pDepthStencilView != nullptr)
+	{
+		com_ptr<ID3D11Texture2D> texture;
+		com_ptr<ID3D11Resource> resource;
+		D3D11_TEXTURE2D_DESC texture_desc;
+
+		pDepthStencilView->GetResource(&resource);
+		if (!FAILED(resource->QueryInterface(&texture)))
+		{
+			texture->GetDesc(&texture_desc);
+		}
+		else
+		{
+			return;
+		}
+
+		if (!check_depthstencil(texture, texture_desc))
```

Comment:
Both calls to `check_depthstencil` have the same boilerplate code right in front of them (`GetResource` and `GetDesc`). Makes sense to move that code inside the function to avoid code duplication.

Reply:
This is because i need them (texture and texture_desc) in the rest of the methods set_active_OM_depthstencil and set_active_cleared_depthstencil